SimPEG/directives/pgi_directives.py

```diff
@@ -63,33 +60,26 @@ class PGI_UpdateParameters(InversionDirective):
     keep_ref_fixed_in_Smooth = True  # keep mref fixed in the Smoothness
 
     def initialize(self):
-        if getattr(self.reg.objfcts[0], "objfcts", None) is not None:
-            pgi_reg = np.where(
-                np.r_[
-                    [
-                        isinstance(regpart, (PGI, PGIwithRelationships))
-                        for regpart in self.reg.objfcts
-                    ]
-                ]
-            )[0][0]
-            self.pgi_reg = self.reg.objfcts[pgi_reg]
-            self._regmode = 1
 
-        else:
-            self._regmode = 2
-            self.pgi_reg = self.reg
+        pgi_reg = self.reg.get_functions_of_type(PGIsmallness)
+        if len(pgi_reg) != 1:
+            raise UserWarning(
+                "'PGI_UpdateParameters' requires one 'PGIsmallness' regularization "
+                "in the objective function."
+            )
+        self.pgi_reg = pgi_reg[0]
 
     def endIter(self):
         if self.opt.iter > 0 and self.opt.iter % self.update_rate == 0:
             m = self.invProb.model
             modellist = self.pgi_reg.wiresmap * m
             model = np.c_[[a * b for a, b in zip(self.pgi_reg.maplist, modellist)]].T
 
-            if self.pgi_reg.mrefInSmooth and self.keep_ref_fixed_in_Smooth:
-                self.fixed_membership = np.c_[
-                    np.arange(len(self.pgi_reg.gmmref.cell_volumes)),
-                    self.pgi_reg.compute_quasi_geology_model(),
-                ]
+            # if self.pgi_reg.reference_model_in_smooth and self.keep_ref_fixed_in_Smooth:
+            #     self.fixed_membership = np.c_[
+            #         np.arange(len(self.pgi_reg.gmmref.cell_volumes)),
+            #         self.pgi_reg.compute_quasi_geology_model(),
+            #     ]
```

Comment:
If these lines are intended to be commented out, might be worth just remove them.

Reply:
Done